src/appleseed/foundation/math/sampling/qmcsamplingcontext.h

```diff
@@ -411,20 +379,46 @@ inline Vector<T, N> QMCSamplingContext<RNG>::next2(Tag<Vector<T, N>>)
             }
         }
     }
-    else if (m_mode == CMJMode)
+    else
     {
-        Vector<T, 2> sample_pos;
-        for (size_t i = 0; i < N; i += 2)
+        for (size_t i = 0; i < N; ++i)
+            v[i] = rand2<T>(m_rng);
+    }
+
+    ++m_instance;
+
+    return v;
+}
+
+template <typename RNG>
+template <typename T>
+inline std::vector<Vector<T, 2>> QMCSamplingContext<RNG>::next2(const size_t N)
+{
+	std::vector<Vector<T, 2>> v(N);
+
+    assert(m_sample_count == 0 || m_instance < m_sample_count);
+    assert(m_dimension == 2);
+    assert(N <= PrimeTableSize);
```

Comment:
Can you explain me this assert?

Reply:
The assert on N is to check if N can be found in the precomputed primes table. But we don't use it here so I can remove it.
For the dimension, not required but it makes sens to me. If you create a sampling context for 1D you don't it to work with 2D.

Reply:
All unused code should be removed;